internal/v4/auth_test.go

```diff
@@ -726,3 +727,119 @@ func (s *authSuite) TestIsEntityCaveat(c *gc.C) {
 		c.Assert(rec.Code, gc.Equals, http.StatusOK)
 	}
 }
+
+func (s *authSuite) TestDelegatableMacaroon(c *gc.C) {
+	// Create a new server with a third party discharger.
+	srv, store, discharger := newServerWithDischarger(c, s.Session, "bob", nil)
+	defer discharger.Close()
+
+	// First check that we get a macaraq error when using a vanilla http do
+	// request.
+	httptesting.AssertJSONCall(c, httptesting.JSONCallParams{
+		Handler: srv,
+		URL:     storeURL("macaroon-delegatable"),
+		ExpectBody: httptesting.BodyAsserter(func(c *gc.C, m json.RawMessage) {
+			// Allow any body - the next check will check that it's a valid macaroon.
+		}),
+		ExpectStatus: http.StatusProxyAuthRequired,
+	})
+
+	client := httpbakery.NewHTTPClient()
+
+	now := time.Now()
+	var gotBody json.RawMessage
+	httptesting.AssertJSONCall(c, httptesting.JSONCallParams{
+		Handler: srv,
+		URL:     storeURL("macaroon-delegatable"),
+		ExpectBody: httptesting.BodyAsserter(func(c *gc.C, m json.RawMessage) {
+			gotBody = m
+		}),
+		Do: func(req *http.Request) (*http.Response, error) {
+			return httpbakery.Do(client, req, noInteraction)
+		},
+		ExpectStatus: http.StatusOK,
+	})
+
+	c.Assert(gotBody, gc.NotNil)
+	var m macaroon.Macaroon
+	err := json.Unmarshal(gotBody, &m)
+	c.Assert(err, gc.IsNil)
+
+	caveats := m.Caveats()
+	foundExpiry := false
+	for _, cav := range caveats {
+		cond, arg, err := checkers.ParseCaveat(cav.Id)
+		c.Assert(err, gc.IsNil)
+		switch cond {
+		case checkers.CondTimeBefore:
```

Comment:
They're checked by implication when the authorization is checked. We only want to check the expirty time here because it's not checked any other way.